src/main/java/com/redhat/devtools/lsp4ij/LSPIJUtils.java

```diff
@@ -413,22 +413,40 @@ private static Language doGetFileLanguage(@NotNull VirtualFile file, @NotNull Pr
 
 
     /**
-     * Returns the virtual file corresponding to the Psi file.
+     * Returns the virtual file corresponding to the PSI element.
      *
+     * @param element the PSI element
      * @return the virtual file, or {@code null} if the file exists only in memory.
      */
     public static @Nullable VirtualFile getFile(@NotNull PsiElement element) {
         PsiFile psFile = element.getContainingFile();
         return psFile != null ? psFile.getVirtualFile() : null;
     }
 
+    /**
+     * Returns the document corresponding to the virtual file.
+     *
+     * @param file the virtual file
+     * @return the document corresponding to the virtual file, or {@code null} if no document could be found
+     */
     public static @Nullable Document getDocument(@NotNull VirtualFile file) {
         if (ApplicationManager.getApplication().isReadAccessAllowed()) {
             return FileDocumentManager.getInstance().getDocument(file);
         }
         return ReadAction.compute(() -> FileDocumentManager.getInstance().getDocument(file));
     }
 
+    /**
+     * Returns the document corresponding to the PSI element
+     *
+     * @param element the PSI element
+     * @return the document corresponding to the PSI element, or {@code null} if no document could be found
+     */
+    public static @Nullable Document getDocument(@NotNull PsiElement element) {
```

Comment:
Added this composite signature for convenience of translation directly from a PSI element to the corresponding document.